### ISSUE TITLE
Unable to focus on TextField or open the keyboard again after side pop from another screen

### ISSUE THREAD

**Issue by arabnology** [flutter/flutter#148867]
### Steps to reproduce

I have a TextField on screen 1. When I go to a new screen using "Navigator.of(context).push(new Screen)" and then using swipe on the new screen side to pop (iOS), I'm unable to click on the TextField again, and the keyboard won't open either.

Note: If i use "Navigator.of(context).pop()" to go back to screen 1, it will work as expected, however, the issue only appears when using Swip on the new screen side to pop back to screen 1.

Tested on iOS only.

### Expected results

Be able to focus on the TextField again and to open the keyboard also.

### Actual results

Unable to focus on TextField or open the keyboard again.

### Code sample

<details open><summary>Code sample</summary>

```dart
PopScope(
        canPop: true,
        child: Scaffold(),
),
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/88229381/6f37174a-d391-4697-ab1a-74e46fe6037b

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.5 23F79 darwin-x64, locale
    en-SA)
    • Flutter version 3.22.0 on channel stable at /Users/mimac/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (13 days ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/mimac/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.10+0-17.0.10b1087.21-11572160)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15F31d
    • CocoaPods version 1.15.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /Applications/Android Studio.app
    • Java version OpenJDK Runtime Environment (build
      17.0.10+0-17.0.10b1087.21-11572160)

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (4 available)
    • iPad (mobile)                                  • 00008020-000D70D622BA402E            • ios            • iOS 17.5.1 21F90
    • iPad Pro (12.9-inch) (6th generation) (mobile) • 6E203326-1177-4E30-A1E6-EE8FAEF3783E • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)                                • macos                                • darwin-x64     • macOS 14.5 23F79
      darwin-x64
    • Chrome (web)                                   • chrome                               • web-javascript • Google Chrome
      125.0.6422.76

[!] Network resources
    ✗ A network error occurred while checking "https://github.com/": Operation timed out

! Doctor found issues in 1 category.
```

</details>

**Comment by KorayLiman** [comment]
I have the same issue, after popping from a page can't focus any textformfield.

**Comment by darshankawar** [comment]
Thanks for the report @arabnology 
Is it using go_router or just navigation ?
Can you also check below issues and see if they resemble your case or not ?

https://github.com/flutter/flutter/issues/138624
https://github.com/flutter/flutter/issues/120353
https://github.com/flutter/flutter/issues/141268

**Comment by wasilni2022** [comment]
Same here, I have the same issue (iOS, Using flutter built in Navigator)

**Comment by MrOlolo** [comment]
We have the same problem at Flutter 3.22.0

**Comment by mlshvdv** [comment]
We also have the same. 3.22.0

**Comment by darshankawar** [comment]
Thanks for the updates. I was able to replicate the reported behavior using https://docs.flutter.dev/cookbook/navigation/named-routes#interactive-example in latest stable version but not in master version, as shown below:

`Channel master, 3.22.0-45.0.pre`


https://github.com/flutter/flutter/assets/67046386/5c2da5c4-f0c8-422c-b913-6c9a5eb21193

Indicating it is likely fixed in master. Everyone who reported / facing this issue, please switch to latest master and re-run your scenario to confirm.

**Comment by techouse** [comment]
I was able to replicate this on v3.22.1.

Oddly enough, this is only an issue on iOS when you swipe back, not when you tap the back button.

**Comment by MrOlolo** [comment]
> re-run your scenario to confirm.

Hey! Checked at `Channel master, 3.22.0-45.0.pre` and everything works fine - no problem with textfied & keyboard

**Comment by arabnology** [comment]
Ok, I will wait for the next stable version to be released. Thank you.

**Comment by darshankawar** [comment]
Closing this as fixed, since it works as expected in master version.

**Comment by MrOlolo** [comment]
> Closing this as fixed, since it works as expected in master version.

So it will be in the next hotfix release?

**Comment by jonmountjoy** [comment]
Apparently it won't be a hotfix @MrOlolo @darshankawar  - see https://github.com/flutter/flutter/issues/148573#issuecomment-2128483148 - which is really surprising given that it's a big regression.

**Comment by MrOlolo** [comment]
> Apparently it won't be a hotfix @MrOlolo @darshankawar  - see https://github.com/flutter/flutter/issues/148573#issuecomment-2128483148 - which is really surprising given that it's a big regression.

OMG. This is total trash. It is a terrible regression and iPhone owners will destroy us if we leave this problem at our app

**Comment by DeanFs** [comment]
> Thanks for the updates. I was able to replicate the reported behavior using https://docs.flutter.dev/cookbook/navigation/named-routes#interactive-example in latest stable version but not in master version, as shown below:
> 
> `Channel master, 3.22.0-45.0.pre`
> 
>  148867_m.mov 
> Indicating it is likely fixed in master. Everyone who reported / facing this issue, please switch to latest master and re-run your scenario to confirm.

After I switched to Channel master, 3.22.0-45.0.pre, extended_text was incompatible and an exception was thrown during compilation: "Error (Xcode): ../../../.pub-cache/hosted/pub-web .flutter-io.cn/extended_text-13.0.0/lib/src/extended/text_overflow_mixin.dart:882:31: Error: The method 'TextOverflowMixin.layoutInlineChildren' has fewer positional arguments than those of overridden method 'RenderBox with ContainerRenderObjectMixin, RenderInlineChildrenContainerDefaults.layoutInlineChildren'."
I see there is no new version available for extended_text

**Comment by kecson** [comment]
when will update the hotfix release?

**Comment by jmagman** [comment]
Thank you, @techouse !

**Comment by BazinC** [comment]
@darshankawar @justinmc @chunhtai any eta for a hotfix for stable channel? 🙏
This bug is pretty annoying and we cannot ship new version of our app.
Reverting back to 3.19 would require a lot of effort.
Thank you

**Comment by ebwood** [comment]
> > > Apparently it won't be a hotfix
> > 
> > 
> > > If you can find the related PR that fixed the 1st issue, you can submit a [Request a cherry-pick](https://github.com/flutter/flutter/issues/new?assignees=&labels=cp%3A+review&projects=&template=7_cherry_pick.yml&title=%5BCP%5D+%3Ctitle%3E).
> > 
> > 
> > It's not that it won't be a hotfix, it's that it's unclear which commit should be hotfixed, which would be necessary to make the hotfix. If you're interesting in this being fixed in stable, the most helpful thing would be to git bisect your Flutter SDK from when you knew it didn't work (stable) until you know it doesn't work (master), and then follow the instructions in the link to hotfix request instructions, or at least reply and tell us what you found.
> 
> @jmagman I started digging and found out that this was fixed in [3.22.0-0.19.0.pre](https://github.com/flutter/flutter/releases/tag/3.22.0-19.0.pre) and is broken in [3.22.0-0.18.0.pre](https://github.com/flutter/flutter/releases/tag/3.22.0-18.0.pre).
> 
> I tested these versions
> 
> Version	Works	Comment
> 3.22.0-0.45.0.pre	✅	
> 3.22.1	❌	
> 3.22.0-0.44.0.pre	✅	
> 3.22.0-0.43.0.pre	✅	
> 3.22.0-0.30.0.pre	✅	
> 3.22.0	❌	
> 3.22.0-0.29.0.pre	✅	
> 3.22.0-0.20.0.pre	✅	
> [3.22.0-0.19.0.pre](https://github.com/flutter/flutter/releases/tag/3.22.0-19.0.pre)	✅	fixed here
> 3.22.0-0.18.0.pre	❌	
> 3.22.0-0.17.0.pre	❌	
> I made a cherrypick request here #149178

Does it mean the stable version is not stable anymore?😄

**Comment by wahkim** [comment]
> > > 显然，这不会是一个热修复程序> >> >> > > > If fineltre修复第一个问题的相关PR，您可以提交[请求樱桃选择](https://github.com/flutter/flutter/issues/new?分配人=&labels=cp%3A+review&projects=&template=7_cherry_pick.yml&title=%5BCP%5D+%3Ctitle%3E）。> >> >> > 这不是说它不是修补程序，而是不清楚哪个提交应该修补修，这是制作修补程序所必需的。如果对稳定修复感兴趣，最有帮助的事情从你知道它不起作用（稳定）开始，直到你知道它不起作用（主），然后按照热修复请求说明链接中的说明进行分，或者至少回复并告诉我们你发现了什么。>> @jmagman我开始挖掘，发现这在[3.22.0-0.19.0.pre](https://github.com/flutter/flutter/releases/tag/3.22.0-19.0.pre)中修复，在[3.22.0-0.18.0.pre](https://github.com/flutter/flutter/releases/tag/3.22.0-18.0.pre)中损坏。>> 我测试了这些版本>> 版本	作品	评论> 3.22.0-0.45.0.pre	✅	> 3.22.1	❌	> 3.22.0-0.44.0.pre	✅	> 3.22.0-0.43.0.pre	✅	> 3.22.0-0.30.0.pre	✅	> 3.22.0	❌	> 3.22.0-0.29.0.pre	✅	> 3.22.0-0.20.0.pre	✅	> [3.22.0-0.19.0.pre]（https://github.com/flutter/flutter/releases/tag/3.22.0-19.0.pre）	✅	固定在这里> 3.22.0-0.18.0.pre	❌	> 3.22.0-0.17.0.pre	❌	> 我在这里提出了一个樱桃挑选请求#149178

How to specify update to version 3.22.0-0.19.0.pre

**Comment by ismailcaakir** [comment]
Unfortunately I solved the problem by downgrading to 3.19.0 on the stable channel.

**Comment by jmagman** [comment]
A fix for this was hotfixed as Flutter 3.22.2, please try it out!

**Comment by michaelnew** [comment]
Just tested on 3.22.2 and it works great. Thanks all ♥️
